src/Tribe/Integrations/Freemius.php

```diff
@@ -0,0 +1,386 @@
+	 *
+	 * @since  TBD
+	 *
+	 * @var string
+	 */
+	private $object_class = 'Tribe__Tickets__Main';
+
+	/**
+	 * Stores the public key for Freemius.
+	 *
+	 * @since  TBD
+	 *
+	 * @var string
+	 */
+	private $public_key = 'pk_6dd9310b57c62871c59e58b8e739e';
+
+	/**
+	 * Stores the ID for the Freemius application.
+	 *
+	 * @since  TBD
+	 *
+	 * @var string
+	 */
+	private $freemius_id = '3841';
+
+	/**
+	 * Stores the slug for the Freemius application.
+	 *
+	 * @since  TBD
+	 *
+	 * @var string
+	 */
+	private $slug = 'event-tickets';
+
+	/**
+	 * Stores the name for the Freemius application.
+	 *
+	 * @since  TBD
+	 *
+	 * @var string
+	 */
+	private $name = 'Event Tickets';
+
+	/**
+	 * Store the value from the 'page' in the request.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private $page = 'tribe-common';
+
+	/**
+	 * Tribe__Tickets__Integrations__Freemius constructor.
+	 *
+	 * @since TBD
+	 */
+	public function __construct() {
+		$this->setup();
+	}
+
+	/**
+	 * Performs setup for the Freemius integration singleton.
+	 *
+	 * @since  TBD
+	 */
+	public function setup() {
+		if ( ! is_admin() ) {
+			return;
+		}
+		// Setup possible redirect.
+		add_action( 'wp_loaded', [ $this, 'action_redirect_incorrect_page' ] );
+
+		global $pagenow;
+
+		$page = tribe_get_request_var( 'page' );
+
+		$valid_page = [
+			Tribe__Settings::$parent_slug => true,
+			Tribe__App_Shop::MENU_SLUG    => true,
+			'tribe-help'                  => true,
+		];
+
+		if ( class_exists( 'Tribe__Events__Aggregator__Page' ) ) {
+			$valid_page[ Tribe__Events__Aggregator__Page::$slug ] = true;
+		}
+
+		if ( isset( $valid_page[ $page ] ) ) {
+			$this->page = $page;
+		} elseif ( 'plugins.php' !== $pagenow && ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) ) {
+			return;
+		}
+
+		// If the common that loaded doesn't include Freemius, let's bail.
+		if ( ! tribe()->offsetExists( 'freemius' ) ) {
+			return;
+		}
+
+		$should_load = true;
+
+		// Check if Freemius integration is disabled.
+		if ( ( defined( 'TRIBE_NO_FREEMIUS' ) && true === TRIBE_NO_FREEMIUS ) || true === (bool) getenv( 'TRIBE_NO_FREEMIUS' ) ) {
+			$should_load = false;
+		}
+
+		/**
+		 * Allows third-party disabling of the integration.
+		 *
+		 * @since  TBD
+		 *
+		 * @param bool $should_load Whether the Freemius integration should load.
+		 */
+		$should_load = apply_filters( 'tribe_tickets_integrations_should_load_freemius', $should_load );
+
+		if ( ! $should_load ) {
+			return;
+		}
+
+		/** @var Tribe__Freemius $freemius */
+		$freemius = tribe( 'freemius' );
+
+		$this->instance = $freemius->initialize( $this->slug, $this->freemius_id, $this->public_key, [
+			'menu'           => [
+				'slug'       => $this->page,
+				'first-path' => $this->get_settings_path(),
+				'account'    => false,
+				'contact'    => false,
+				'support'    => false,
+			],
+			'is_premium'     => false,
+			'has_addons'     => false,
+			'has_paid_plans' => false,
+		] );
+
```

Comment:
:no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: d3b69512ddef750e95696e0648ac14cd</sub>